sdc-bareminimum/project-catwalk-related-service
# test endpoints with Postman and k6

Update docker compose files

Hide Remote EC2 Postgres connection string with username and password.
Update the Node Server to connect to the remote database.

Added support the the "k6" load testing tool.
Added `k6-script.js` to test EACH endpoint and ensure latency < 2000ms.
Created a Postman Collection for test each endpoint's response code and latency.
Tested URLs for data in last 10 of data set.

[Article](https://medium.com/@gururajhm/load-testing-api-with-postman-node-js-k6-8fa04725e3b4)
```js
// In Postman
METHOD: GET http://localhost:3000/products/

// Tests tab
pm.test("Status code is 200", function () {
  pm.response.to.have.status(200);
});

pm.test("Response time is less than 2000ms", () => {
  pm.expect(pm.response.responseTime).to.be.below(2000);
});
```

```sh
* Create Postman collection *
* Export Postman collection to .json file*
$ npm install -g postman-to-k6
$ postman-to-k6 Capsules_Testpostman_collection.json — iterations 25 -o k6-script.js
$ k6 run k6-script.js
$ k6 login cloud --token <YOUR_K6_TOKEN>
$ k6 cloud k6-script.js
```
